src/pepr/operator/controllers/network/policies.ts

```diff
@@ -17,6 +18,37 @@ import { defaultDenyAll } from "./defaults/default-deny-all";
 import { generate } from "./generate";
 import { allowAmbientHealthprobes } from "./generators/ambientHealthprobes";
 
+/**
+ * Finds the SSO client that matches the given service selector
+ */
+export function findMatchingClient(pkg: UDSPackage, serviceSelector: Record<string, string>) {
```

Comment:
nit: this is a pod label/selector, the `serviceSelector` confused me at first when tracing through the code here